morphir/src/org/finos/morphir/runtime/RTValue.scala

```diff
@@ -256,6 +262,80 @@ object RTValue {
     )
   }
 
+  object Order {
+    val GT = ConstructorResult(FQName.fromString("Morphir.SDK:Basics:GT"), scala.List.empty)
+    val LT = ConstructorResult(FQName.fromString("Morphir.SDK:Basics:LT"), scala.List.empty)
+    val EQ = ConstructorResult(FQName.fromString("Morphir.SDK:Basics:EQ"), scala.List.empty)
+  }
+
+  /**
+   * Trait for types that *may* be comparable in Morphir; Note that not everythign that extends this may actually be
+   * compared. For instance, lists are comparable if and only if their elements are comparable.
+   */
+  sealed trait Comparable extends RTValue {
+    final def compare(that: Comparable): Int =
+      Comparable.compareOrThrow(this, that)
+  }
+  object Comparable {
+    def compare(first: Comparable, second: Comparable): Either[IllegalValue, Int] = {
+      def lexicalHelper(first: scala.List[RTValue], second: scala.List[RTValue]) = {
+        val zipped = first.zip(second)
+        val mapped = zipped.map {
+          case (a: Comparable, b: Comparable) => recursiveHelper(a, b)
+          case (a, b: Comparable) =>
+            Left(IllegalValue(s"Cannot compare values $a and $b because the first is not comparable"))
+          case (a: Comparable, b) =>
+            Left(IllegalValue(s"Cannot compare values $a and $b because the second is not comparable"))
+          case (a, b) => Left(IllegalValue(s"Cannot compare values $a and $b because neither is comparable"))
+        }
+        mapped.find(_.isLeft) match {
+          case Some(error) => error
+          case None =>
+            val rights       = mapped.map(_.getOrElse(throw new Exception("unreachable branch reached")))
+            val firstNonZero = rights.find(_ != 0)
+            firstNonZero match {
+              case Some(value) => Right(value)
+              case None        => Right(first.length.compare(second.length))
+            }
+        }
+      }
+      // This helper function exists just to neatly wrap leaf errors with context from the top-level comparison that was attempted
+      def recursiveHelper(first: Comparable, second: Comparable): Either[IllegalValue, Int] =
+        (first, second) match {
+          case (Primitive.Int(a), Primitive.Int(b))       => Right(a.compare(b))
+          case (Primitive.Float(a), Primitive.Float(b))   => Right(a.compare(b))
+          case (Primitive.Char(a), Primitive.Char(b))     => Right(a.compare(b))
+          case (Primitive.String(a), Primitive.String(b)) => Right(a.compare(b))
+          case (Tuple(a_elements), Tuple(b_elements)) =>
+            if (a_elements.length == b_elements.length) lexicalHelper(a_elements, b_elements)
+            else Left(IllegalValue(s"Cannot compare tuples $first and $second because they have different lengths"))
+          case (List(a_elements), List(b_elements)) => lexicalHelper(a_elements, b_elements)
```

Comment:
So I'm not sure what you're saying, exactly - you're correct that a list of records is not comparable. This code is not explicitly enforcing that (That's up to the type checker), but in (almost) all cases it should fail with a sane error - attempting to compare two lists of records will fail because it will try to recursively compare the records.

The one illegal thing it will let thru is if you compare an *empty* list of records; preventing that at runtime does require adding the type tag to RTValues, which is high on my list.

I'll add the tests you suggested.